airbyte-oauth/src/main/java/io/airbyte/oauth/flows/PinterestOAuthFlow.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.oauth.flows;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.oauth.BaseOAuth2Flow;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * Following docs from https://developers.pinterest.com/docs/getting-started/authentication
+ */
+public class PinterestOAuthFlow extends BaseOAuth2Flow {
```

Comment:
@arsenlosenko could you add unit tests for this function?

Reply:
@sherifnada yes, already working on unit tests for this function in scope of this PR:
https://github.com/airbytehq/airbyte/pull/16393